### ISSUE TITLE
Loading my own state_dict but gradient becomes zero

### ISSUE THREAD

**Issue by linzhiqiu** [pytorch/pytorch#10737]
I was trying to compute the gradient w.r.t input image using a resnet model. When using pretrained/not trained resnet model, the gradient can be computed correctly; when I instead load a module using 

`model_dict = torch.load(model_path)`
`model.load_state_dict(model_dict)`

The gradient computed then becomes zero w.r.t input.

I am pretty sure my model is working as it gets pretty good accuracy. Any idea what might happen?

**Comment by zhuyw05** [comment]
set the parameter keep_vars=True then you got gradients
